src/lib/datatransfers/streamer/streamer_datatransfer.py

```diff
@@ -0,0 +1,170 @@
+        self.work_dir = work_dir
+        self.system_name = system_name
+        self.pypi_index_url = pypi_index_url
+        self.port_range = port_range
+        self.public_ips = public_ips if public_ips else ["localhost"]
+        self.host = host
+        self.wait_timeout = wait_timeout
+        self.inbound_transfer_limit = inbound_transfer_limit
+
+    async def upload(
+        self,
+        source: DataTransferLocation,
+        target: DataTransferLocation,
+        username,
+        access_token,
+        account,
+    ) -> DataTransferOperation | None:
+
+        job_id = None
+        secret = secrets.token_urlsafe(16)
+        start_port, end_port = self.port_range
+        parameters = {
+            "sbatch_directives": _format_directives(self.directives, account),
+            "operation": "receive",
+            "target_path": target.path,
+            "secret": secret,
+            "port_range": f"{start_port} {end_port}",
+            "public_ips": self.public_ips,
+            "host": self.host,
+            "pypi_index_url": self.pypi_index_url,
+            "wait_timeout": self.wait_timeout,
+            "inbound_transfer_limit": self.inbound_transfer_limit,
+        }
+
+        job_script = _build_script("job_streamer.sh", parameters)
+        job = JobHelper(
+            f"{self.work_dir}/{username}", job_script, "IngressFileTransfer"
+        )
+
+        job_id = await self.scheduler_client.submit_job(
+            job_description=JobDescriptionModel(**job.job_param),
+            username=username,
+            jwt_token=access_token,
+        )
+
+        transferJob = TransferJob(
+            job_id=job_id,
+            system=target.system,
+            working_directory=job.working_dir,
+            logs=TransferJobLogs(
+                output_log=job.job_param["standard_output"],
+                error_log=job.job_param["standard_error"],
+            ),
+        )
+
+        coordinates = {
+            "ports": [start_port, end_port],
+            "ips": self.public_ips,
+            "secret": secret,
+        }
+        encoded = base64.urlsafe_b64encode(
+            json.dumps(coordinates).encode("utf-8")
+        ).decode("utf-8")
+
+        directives = StreamerDataTransferDirective(
+            **{"coordinates": encoded, "transfer_method": "streamer"}
+        )
+
+        return StreamerDataTransferOperation(
+            transferJob=transferJob,
+            transfer_directives=directives,
+        )
+
+    async def download(
+        self,
+        source: DataTransferLocation,
+        target: DataTransferLocation,
+        username,
+        access_token,
+        account,
+    ) -> DataTransferOperation | None:
+
+        job_id = None
+        secret = secrets.token_urlsafe(16)
+        start_port, end_port = self.port_range
+        parameters = {
+            "sbatch_directives": _format_directives(self.directives, account),
```

Comment:
`host` parameter is missing, resulting in an error when launching the server on download task via scheduler:
```
Thu Oct 16 10:42:04 UTC 2025 Starting firecrest streamer in send mode...
Usage: streamer server [OPTIONS] COMMAND [ARGS]...
Try 'streamer server --help' for help.

Error: No such command 'localhost'.
```
```suggestion
            "sbatch_directives": _format_directives(self.directives, account),
            "host": self.host,
```

Reply:
oh sorry about that! Thanks, fixed